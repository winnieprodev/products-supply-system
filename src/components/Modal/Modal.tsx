import { CloseIcon } from "components";
import React from "react";

import * as Styled from "./modal.styled";

interface OwnProps {
  isOpen: boolean;
  centered?: boolean;
  size?: Styled.ModalSize;
  onClose: (isOpen: boolean) => void;
}

const Modal: React.FC<React.PropsWithChildren<OwnProps>> = ({
  isOpen,
  centered = false,
  size = "medium",
  children,
  onClose,
}) => {
  const toggleIsOpen = () => onClose(!isOpen);
  return (
    <Styled.Modal isOpen={isOpen}>
      <Styled.ModalWrapper $size={size} $centered={centered}>
        <Styled.ModalContent>
          <Styled.CloseIcon onClick={toggleIsOpen}>
            <CloseIcon />
          </Styled.CloseIcon>
          {children}
        </Styled.ModalContent>
      </Styled.ModalWrapper>
      <Styled.CancelArea onClick={toggleIsOpen} />
    </Styled.Modal>
  );
};

export default React.memo(Modal);
