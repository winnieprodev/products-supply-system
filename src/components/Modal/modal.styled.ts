import styled, { css, keyframes } from "styled-components";
import { secondaryColorLighter } from "styles/colors";
export type ModalSize = "small" | "medium" | "large";

interface ModalCoreProps {
  isOpen: boolean;
}

interface ModalWrapperProps {
  $centered: boolean;
  $size: ModalSize;
}

export const Modal = styled.div<ModalCoreProps>`
  position: fixed;
  top: 0;
  left: 0;
  display: ${({ isOpen }) => (isOpen ? "block" : "none")};
  z-index: +1;
  width: 100%;
  height: 100%;
`;

export const CloseIcon = styled.div`
  cursor: pointer;
  position: absolute;
  top: 0;
  right: 0;
  font-size: 24px;
  width: 24px;
  text-align: center;
  padding: 8px;
  border-radius: 50%;
  color: #000000;
  &:hover {
    background: ${secondaryColorLighter};
  }
`;

const openDropdownAnimation = keyframes`
    0% { 
        transform: scale(0.1) translateY(-30px) translateX(150px); 
        opacity: 0.1; 
    }
    100% {
        transform: scale(1) translateY(0) translatex(0);
        opacity: 1;
    }
`;

export const ModalWrapper = styled.div<ModalWrapperProps>`
  transform: none;
  position: relative;
  display: flex;
  align-items: center;
  margin: 1.75rem auto;
  pointer-events: none;
  border-radius: 6px;
  animation: ${openDropdownAnimation} ease 0.25s;
  width: auto;
  min-width: 360px;
  ${({ $centered }) =>
    $centered &&
    css`
      min-height: calc(100% - 3.5rem);
    `}
  ${({ $size }) =>
    $size === "small"
      ? css`
          max-width: 400px;
        `
      : $size === "medium"
      ? css`
          max-width: 600px;
        `
      : $size === "large" &&
        css`
          max-width: 900px;
        `}
`;

export const CancelArea = styled.div`
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background: rgba(255, 255, 255, 0.8);
  z-index: -1;
`;

export const ModalContent = styled.div`
  position: relative;
  display: flex;
  flex-direction: column;
  box-shadow: 0 1px 4px rgba(0, 0, 0, 0.15), 0 4px 10px rgba(0, 0, 0, 0.15);
  width: 100%;
  padding: 16px 24px;
  pointer-events: auto;
  background-color: #fff;
  background-clip: padding-box;
  border: 0 solid rgba(0, 0, 0, 0.2);
  border-radius: 0.3rem;
  outline: 0;
`;
