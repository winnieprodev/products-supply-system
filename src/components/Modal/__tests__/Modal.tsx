import { render } from "@testing-library/react";

import Modal from "../Modal";

describe("Modal", () => {
  describe("Snapshot tests", () => {
    test("it matches the snapshot with default. props", () => {
      const { container } = render(
        <Modal isOpen={true} onClose={jest.fn()}>
          test
        </Modal>,
      );
      expect(container.firstChild).toMatchSnapshot();
    });
    test("it matches the snapshot with size = small", () => {
      const { container } = render(
        <Modal isOpen={true} size={"small"} onClose={jest.fn()}>
          test
        </Modal>,
      );
      expect(container.firstChild).toMatchSnapshot();
    });
    test("it matches the snapshot with size = medium", () => {
      const { container } = render(
        <Modal isOpen={true} size={"medium"} onClose={jest.fn()}>
          test
        </Modal>,
      );
      expect(container.firstChild).toMatchSnapshot();
    });
    test("it matches the snapshot with size = large", () => {
      const { container } = render(
        <Modal isOpen={true} size={"large"} onClose={jest.fn()}>
          test
        </Modal>,
      );
      expect(container.firstChild).toMatchSnapshot();
    });
    test("it matches the snapshot with centered = true", () => {
      const { container } = render(
        <Modal isOpen={true} centered onClose={jest.fn()}>
          test
        </Modal>,
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
