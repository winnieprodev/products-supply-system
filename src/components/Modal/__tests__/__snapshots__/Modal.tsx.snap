// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal Snapshot tests it matches the snapshot with centered = true 1`] = `
.c0 {
  position: fixed;
  top: 0;
  left: 0;
  display: block;
  z-index: +1;
  width: 100%;
  height: 100%;
}

.c3 {
  cursor: pointer;
  position: absolute;
  top: 0;
  right: 0;
  font-size: 24px;
  width: 24px;
  text-align: center;
  padding: 8px;
  border-radius: 50%;
  color: #000000;
}

.c3:hover {
  background: #F3F3F3;
}

.c1 {
  -webkit-transform: none;
  -ms-transform: none;
  transform: none;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 1.75rem auto;
  pointer-events: none;
  border-radius: 6px;
  -webkit-animation: eKCkCK ease 0.25s;
  animation: eKCkCK ease 0.25s;
  width: auto;
  min-width: 360px;
  min-height: calc(100% - 3.5rem);
  max-width: 600px;
}

.c4 {
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background: rgba(255,255,255,0.8);
  z-index: -1;
}

.c2 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-shadow: 0 1px 4px rgba(0,0,0,0.15),0 4px 10px rgba(0,0,0,0.15);
  width: 100%;
  padding: 16px 24px;
  pointer-events: auto;
  background-color: #fff;
  background-clip: padding-box;
  border: 0 solid rgba(0,0,0,0.2);
  border-radius: 0.3rem;
  outline: 0;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      <div
        class="c3"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-times fa-w-11 "
          data-icon="times"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 352 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
            fill="currentColor"
          />
        </svg>
      </div>
      test
    </div>
  </div>
  <div
    class="c4"
  />
</div>
`;

exports[`Modal Snapshot tests it matches the snapshot with default. props 1`] = `
.c0 {
  position: fixed;
  top: 0;
  left: 0;
  display: block;
  z-index: +1;
  width: 100%;
  height: 100%;
}

.c3 {
  cursor: pointer;
  position: absolute;
  top: 0;
  right: 0;
  font-size: 24px;
  width: 24px;
  text-align: center;
  padding: 8px;
  border-radius: 50%;
  color: #000000;
}

.c3:hover {
  background: #F3F3F3;
}

.c1 {
  -webkit-transform: none;
  -ms-transform: none;
  transform: none;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 1.75rem auto;
  pointer-events: none;
  border-radius: 6px;
  -webkit-animation: eKCkCK ease 0.25s;
  animation: eKCkCK ease 0.25s;
  width: auto;
  min-width: 360px;
  max-width: 600px;
}

.c4 {
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background: rgba(255,255,255,0.8);
  z-index: -1;
}

.c2 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-shadow: 0 1px 4px rgba(0,0,0,0.15),0 4px 10px rgba(0,0,0,0.15);
  width: 100%;
  padding: 16px 24px;
  pointer-events: auto;
  background-color: #fff;
  background-clip: padding-box;
  border: 0 solid rgba(0,0,0,0.2);
  border-radius: 0.3rem;
  outline: 0;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      <div
        class="c3"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-times fa-w-11 "
          data-icon="times"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 352 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
            fill="currentColor"
          />
        </svg>
      </div>
      test
    </div>
  </div>
  <div
    class="c4"
  />
</div>
`;

exports[`Modal Snapshot tests it matches the snapshot with size = large 1`] = `
.c0 {
  position: fixed;
  top: 0;
  left: 0;
  display: block;
  z-index: +1;
  width: 100%;
  height: 100%;
}

.c3 {
  cursor: pointer;
  position: absolute;
  top: 0;
  right: 0;
  font-size: 24px;
  width: 24px;
  text-align: center;
  padding: 8px;
  border-radius: 50%;
  color: #000000;
}

.c3:hover {
  background: #F3F3F3;
}

.c1 {
  -webkit-transform: none;
  -ms-transform: none;
  transform: none;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 1.75rem auto;
  pointer-events: none;
  border-radius: 6px;
  -webkit-animation: eKCkCK ease 0.25s;
  animation: eKCkCK ease 0.25s;
  width: auto;
  min-width: 360px;
  max-width: 900px;
}

.c4 {
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background: rgba(255,255,255,0.8);
  z-index: -1;
}

.c2 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-shadow: 0 1px 4px rgba(0,0,0,0.15),0 4px 10px rgba(0,0,0,0.15);
  width: 100%;
  padding: 16px 24px;
  pointer-events: auto;
  background-color: #fff;
  background-clip: padding-box;
  border: 0 solid rgba(0,0,0,0.2);
  border-radius: 0.3rem;
  outline: 0;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      <div
        class="c3"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-times fa-w-11 "
          data-icon="times"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 352 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
            fill="currentColor"
          />
        </svg>
      </div>
      test
    </div>
  </div>
  <div
    class="c4"
  />
</div>
`;

exports[`Modal Snapshot tests it matches the snapshot with size = medium 1`] = `
.c0 {
  position: fixed;
  top: 0;
  left: 0;
  display: block;
  z-index: +1;
  width: 100%;
  height: 100%;
}

.c3 {
  cursor: pointer;
  position: absolute;
  top: 0;
  right: 0;
  font-size: 24px;
  width: 24px;
  text-align: center;
  padding: 8px;
  border-radius: 50%;
  color: #000000;
}

.c3:hover {
  background: #F3F3F3;
}

.c1 {
  -webkit-transform: none;
  -ms-transform: none;
  transform: none;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 1.75rem auto;
  pointer-events: none;
  border-radius: 6px;
  -webkit-animation: eKCkCK ease 0.25s;
  animation: eKCkCK ease 0.25s;
  width: auto;
  min-width: 360px;
  max-width: 600px;
}

.c4 {
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background: rgba(255,255,255,0.8);
  z-index: -1;
}

.c2 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-shadow: 0 1px 4px rgba(0,0,0,0.15),0 4px 10px rgba(0,0,0,0.15);
  width: 100%;
  padding: 16px 24px;
  pointer-events: auto;
  background-color: #fff;
  background-clip: padding-box;
  border: 0 solid rgba(0,0,0,0.2);
  border-radius: 0.3rem;
  outline: 0;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      <div
        class="c3"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-times fa-w-11 "
          data-icon="times"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 352 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
            fill="currentColor"
          />
        </svg>
      </div>
      test
    </div>
  </div>
  <div
    class="c4"
  />
</div>
`;

exports[`Modal Snapshot tests it matches the snapshot with size = small 1`] = `
.c0 {
  position: fixed;
  top: 0;
  left: 0;
  display: block;
  z-index: +1;
  width: 100%;
  height: 100%;
}

.c3 {
  cursor: pointer;
  position: absolute;
  top: 0;
  right: 0;
  font-size: 24px;
  width: 24px;
  text-align: center;
  padding: 8px;
  border-radius: 50%;
  color: #000000;
}

.c3:hover {
  background: #F3F3F3;
}

.c1 {
  -webkit-transform: none;
  -ms-transform: none;
  transform: none;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 1.75rem auto;
  pointer-events: none;
  border-radius: 6px;
  -webkit-animation: eKCkCK ease 0.25s;
  animation: eKCkCK ease 0.25s;
  width: auto;
  min-width: 360px;
  max-width: 400px;
}

.c4 {
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background: rgba(255,255,255,0.8);
  z-index: -1;
}

.c2 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-shadow: 0 1px 4px rgba(0,0,0,0.15),0 4px 10px rgba(0,0,0,0.15);
  width: 100%;
  padding: 16px 24px;
  pointer-events: auto;
  background-color: #fff;
  background-clip: padding-box;
  border: 0 solid rgba(0,0,0,0.2);
  border-radius: 0.3rem;
  outline: 0;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      <div
        class="c3"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-times fa-w-11 "
          data-icon="times"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 352 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
            fill="currentColor"
          />
        </svg>
      </div>
      test
    </div>
  </div>
  <div
    class="c4"
  />
</div>
`;
